File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        let maxSequenceLength = 0;
        const numsSet = new Set(nums);

        for (const n of nums) {
            const isNStartOfASequence = !numsSet.has(n - 1);
            if (isNStartOfASequence) {
                let currentElement = n;
                let sequenceLength = 1;

                while (numsSet.has(currentElement + 1)) {
                    currentElement++;
                    sequenceLength++;
                }

                maxSequenceLength = Math.max(maxSequenceLength, sequenceLength);
            }
        }

        return maxSequenceLength;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-3.java


class Solution {
    public int maxProfit(int[] prices) {
        var lowestPrice = prices[0];
        var maxProfit = 0;

        for(var p : prices) {
            lowestPrice = Math.min(lowestPrice, p);
            maxProfit = Math.max(maxProfit, p - lowestPrice);
        }

        return maxProfit;
    }
}
